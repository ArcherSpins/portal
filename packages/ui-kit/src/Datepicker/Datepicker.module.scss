@import '../variables.scss';
@import '../mixins.scss';

.datepicker {
  display: inline-block;
  width: 100%;
  position: relative;

  &.error_picker {
    i, label {
      color: $danger-400 !important;
    }

    ::-webkit-input-placeholder {
      color: $danger-400 !important;
    }
    :-ms-input-placeholder {
      color: $danger-400 !important;
    }
    ::-moz-placeholder {
      color: $danger-400 !important;
    }
    :-moz-placeholder {
      color: $danger-400 !important;
    }
  }

  label {
    display: block;
    margin-bottom: .4rem;
  }
}

.input {
  position: relative;
  outline: none;
  background-color: white;
  input {
    font-size: 1rem;
    flex: 1;
    width: 100%;
  }
}

.label {
  display: inline-block;
  margin-bottom: 0.25rem;
}



.icon__wrap {
  background: rgba($text-color-50, .2);
  align-items: center;
  display: inline-flex;
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  margin: 1px;
  border-left: 1px solid $text-color-50;
  cursor: pointer;
  outline: none;
  @include default-transition(all);
  &:hover {
    background: rgba($text-color-50, .5);
    color: $text-color;
  }
  i {
    font-size: 1em;
    color: $text-color-50;
    padding: 0 0.25em;
  }
}


.overlay {
  font: inherit;
}