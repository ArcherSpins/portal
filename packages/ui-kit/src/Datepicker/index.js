// @flow
import React from 'react';
import DayPickerInput from 'react-day-picker/lib/src/DayPickerInput';
import { DateUtils } from 'react-day-picker';
import { format as dateFnsFormat, parse as dateFnsParse } from 'date-fns';
import classNames from 'classnames';
import Input from '../Input/index';
import styles from './Datepicker.module.scss';
import overlayStyles from './Overlay.module.scss';
// $FlowFixMe
import 'react-day-picker/lib/style.css';


const DEFAULT_FORMAT = 'd.L.y';

type OverlayAlign = 'left' | 'right';

type Props = {
  onDayChange: Date => void,
  format?: string,
  placeholder?: string,
  overlayAlign?: OverlayAlign,
  value?: Date,
  label?: string,
  className?: string,
  name: string,
  label?: string,
  style?: {
    [string]: mixed
  },
  containerProps?: {
    [string]: any
  },
  error?: boolean
};

type NavbarElementProps = {
  onNextClick: () => void,
  onPreviousClick: () => void,
}

const weekdaysShort = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];

const Navbar = ({ onNextClick, onPreviousClick }: NavbarElementProps) => (
  <div className={classNames(overlayStyles.nav)}>
    <button
      type="button"
      onClick={() => onPreviousClick()}
    >
      <i className="icon-left-open" />
    </button>
    <button
      type="button"
      onClick={() => onNextClick()}
    >
      <i className="icon-right-open" />
    </button>
  </div>
);

const DateInput = (props: any) => {
  const { onFocus, onBlur } = props;
  console.log(props);
  return (
    <div
      className={styles.input}
    >
      <Input
        mask={[/\d/, /\d/, '.', /\d/, /\d/, '.', /\d/, /\d/, /\d/, /\d/]}
        {...props}
      />
      <button type="button" onClick={onFocus} onBlur={onBlur} className={styles.icon__wrap}>
        <i className="icon-calendar" />
      </button>
    </div>
  );
};

const getDate = (date: Date, format: string) => dateFnsFormat(date, format);

const parseDate = (str, format) => {
  const parsed = dateFnsParse(str, format, new Date());
  if (DateUtils.isDate(parsed)) {
    return parsed;
  }
  return undefined;
};

const Datepicker = ({
  onDayChange,
  format = DEFAULT_FORMAT,
  placeholder = DEFAULT_FORMAT,
  value,
  className,
  name,
  label,
  style,
  error,
  containerProps,
  overlayAlign,
  ...props
}: Props) => {
  const inputRef = React.createRef();
  return (
    <div
      {...containerProps}
      style={style}
      className={classNames(styles.datepicker, { [styles.error_picker]: error }, className)}
    >
      <label htmlFor={name}>{label}</label>
      {/* <span className={styles.label}>{label}</span> */}
      <DayPickerInput
        name={name}
        dayPickerProps={{
          ...props,
          classNames: {
            ...overlayStyles,
            container: overlayAlign === 'left'
              ? overlayStyles['container_align-left'] : overlayStyles.container,
          },
          weekdaysShort,
          navbarElement: Navbar,
          selectedDays: value,
          onBlur: (e: SyntheticMouseEvent<HTMLElement>) => {
            // hack for hiding day picker
            // see: https://github.com/gpbl/react-day-picker/issues/926

            if (inputRef && inputRef.current) {
              if (!e.relatedTarget) {
                inputRef.current.hideAfterDayClick();
              }
            }
          },
        }}
        ref={inputRef}
        inputProps={{ ref: null }}
        component={DateInput}
        classNames={styles}
        format={format}
        placeholder={placeholder}
        formatDate={getDate}
        parseDate={parseDate}
        onDayChange={onDayChange}
        value={value}
      />
    </div>
  );
};

Datepicker.defaultProps = {
  format: DEFAULT_FORMAT,
  placeholder: 'DD.MM.YYYY',
  value: '',
  label: '',
  className: '',
  style: {},
  error: false,
  containerProps: {},
  overlayAlign: 'right',
};

export default Datepicker;
