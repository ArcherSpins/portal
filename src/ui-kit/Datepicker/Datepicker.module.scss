@import '../variables.scss';
@import '../mixins.scss';

.datepicker {
  display: inline-block;
  width: 100%;
  position: relative;
}

.input {
  display: flex;
  position: relative;
  input {
    border: 1px solid $text-color-50;
    border-radius: 0.25em;
    padding: 0.35em 0.5em;
    font-size: 0.875em;
    flex: 1;
  }
}



.icon__wrap {
  background: rgba($text-color-50, .2);
  align-items: center;
  display: inline-flex;
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  margin: 1px;
  border-left: 1px solid $text-color-50;
  cursor: pointer;
  @include default-transition(all);
  &:hover {
    background: rgba($text-color-50, .5);
    color: $text-color;
  }
  i {
    font-size: 1em;
    color: $text-color-200;
    padding: 0 0.25em;
  }
}


.overlay {
  font: inherit;
}