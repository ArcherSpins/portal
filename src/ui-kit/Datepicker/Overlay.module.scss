@import '../variables.scss';
@import '../mixins.scss';

.container {
  display: inline-block;
  font-size: 1rem;
  position: absolute;
  box-shadow: 0px 2px 23px rgba($text-color, 0.1);
  border-radius: 0.25em;
  background: #fff;
  right: 0;
  margin-top: 0.5rem;
  z-index: 9999999999;
}

.wrapper {
  position: relative;
  flex-direction: row;
  padding-bottom: 1em;

  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
}

.months {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.month {
  display: table;
  margin: 0 1em;
  margin-top: 1em;
  border-spacing: 0;
  border-collapse: collapse;

  -webkit-user-select: none;

     -moz-user-select: none;

      -ms-user-select: none;

          user-select: none;
}

.nav {
  display: inline-block;
  position: absolute;
  right: 1em;
  top: 1em;
  button {
    cursor: pointer;
    background: rgba($text-color-200, 0.1);
    border-radius: 50%;
    width: 2em;
    height: 2em;
    margin-right: 0.5em;
    color: rgba($text-color-200, 0.35);
    font-size: 0.6em;
    @include default-transition(background);
    &:hover {
      background: rgba($text-color-200, 0.25);
    }
  }
}

.navButton {
  position: absolute;
  top: 1em;
  right: 1.5em;
  left: auto;

  display: inline-block;
  margin-top: 2px;
  width: 1.25em;
  height: 1.25em;
  background-position: center;
  background-size: 50%;
  background-repeat: no-repeat;
  color: #8B9898;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
  }
}

.navButtonInteractionDisabled {
  display: none;
}

.caption {
  display: table-caption;
  margin-bottom: 0.5em;
  padding: 0 0.5em;
  text-align: left;
  > div {
    font-weight: 600;
    font-size: 1em;
    color: $text-color-200;
  }
}

.weekdays {
  display: table-header-group;
  margin-top: 1em;
}

.weekdaysRow {
  display: table-row;
}

.weekday {
  display: table-cell;
  padding: 0.5em;
  color: #8B9898;
  text-align: center;
  font-weight: bold;
  
  abbr[title] {
    border-bottom: none;
    text-decoration: none;
  }
}

.body {
  display: table-row-group;
}

.week {
  display: table-row;
}

.day {
  display: table-cell;
  padding: 0.5em;
  border-radius: 50%;
  vertical-align: middle;
  color: $text-color-200;
  text-align: center;
  font-size: 0.875em;
  cursor: pointer;
  @include default-transition(background);
  &:hover {
    background: rgba($main-color,.2);
  }
}

.weekNumber {
  display: table-cell;
  padding: 0.5em;
  min-width: 1em;
  border-right: 1px solid #EAECEC;
  color: #8B9898;
  vertical-align: middle;
  text-align: right;
  font-size: 0.75em;
  cursor: pointer;
}

.interactionDisabled {
  .day {
    cursor: default;
  }
}

.footer {
  padding-top: 0.5em;
}

.todayButton {
  border: none;
  background-color: transparent;
  background-image: none;
  box-shadow: none;
  color: #4A90E2;
  font-size: 0.875em;
  cursor: pointer;
}

/* Default modifiers */

.today {
  color: white;
  background-color: $main-color;
  font-weight: 700;
}

.outside {
  color: #8B9898;
  cursor: default;
}

.disabled {
  color: #DCE0E0;
  cursor: default;
  /* background-color: #eff1f1; */
}

/* Example modifiers */

.DayPicker-Day--sunday {
  background-color: #F7F8F8;
}

.DayPicker-Day--sunday:not(.DayPicker-Day--today) {
  color: #DCE0E0;
}

.DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
  position: relative;

  background-color: #4A90E2;
  color: #F0F8FF;
}

.DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside):hover {
  background-color: #51A0FA;
}

.DayPicker:not(.DayPicker--interactionDisabled)
  .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover {
  background-color: #F0F8FF;
}

/* DayPickerInput */

.DayPickerInput {
  display: inline-block;
}

.DayPickerInput-OverlayWrapper {
  position: relative;
}

.DayPickerInput-Overlay {
  position: absolute;
  left: 0;
  z-index: 1;

  background: white;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.15);
}