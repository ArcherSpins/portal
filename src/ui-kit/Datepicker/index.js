// @flow
import React from 'react';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import { DateUtils } from 'react-day-picker';
import { format as dateFnsFormat, parse as dateFnsParse } from 'date-fns';

import classNames from 'classnames';
// $FlowFixMe
import 'react-day-picker/lib/style.css';

import { Input } from 'ui-kit';
import styles from './Datepicker.module.scss';
import overlayStyles from './Overlay.module.scss';

const DEFAULT_FORMAT = 'd.L.y';

type Props = {
  onDayChange: Date => void,
  format?: string,
  placeholder?: string,
  value?: Date,
  label?: string,
  className?: string,
  name: string,
  label?: string,
  style?: {
    [string]: mixed
  }
};

type NavbarElementProps = {
  onNextClick: () => void,
  onPreviousClick: () => void
}

const weekdaysShort = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];

const Navbar = ({ onNextClick, onPreviousClick }: NavbarElementProps) => (
  <div className={overlayStyles.nav}>
    <button
      type="button"
      onClick={() => onPreviousClick()}
    >
      <i className="icon-left-open" />
    </button>
    <button
      type="button"
      onClick={() => onNextClick()}
    >
      <i className="icon-right-open" />
    </button>
  </div>
);

const DateInput = (props: any) => {
  const { onFocus, onBlur } = props;

  return (
    <div
      className={styles.input}
    >
      <Input
        {...props}
      />
      <button type="button" onClick={onFocus} onBlur={onBlur} className={styles.icon__wrap}>
        <i className="icon-calendar" />
      </button>
    </div>
  );
};

const getDate = (date: Date, format: string) => dateFnsFormat(date, format);

const parseDate = (str, format) => {
  const parsed = dateFnsParse(str, format, new Date());
  if (DateUtils.isDate(parsed)) {
    return parsed;
  }
  return undefined;
};

const Datepicker = ({
  onDayChange,
  format = DEFAULT_FORMAT,
  placeholder = DEFAULT_FORMAT,
  value,
  className,
  name,
  label,
  style,
  ...props
}: Props) => {
  const inputRef = React.createRef();
  return (
    <div style={style} className={classNames(styles.datepicker, className)}>
      <label htmlFor={name}>{label}</label>
      {/* <span className={styles.label}>{label}</span> */}
      <DayPickerInput
        name={name}
        dayPickerProps={{
          ...props,
          classNames: overlayStyles,
          weekdaysShort,
          navbarElement: Navbar,
          selectedDays: value,
          onBlur: (e: SyntheticMouseEvent<HTMLElement>) => {
            // hack for hiding day picker
            // see: https://github.com/gpbl/react-day-picker/issues/926

            if (inputRef && inputRef.current) {
              if (!e.relatedTarget) {
                inputRef.current.hideAfterDayClick();
              }
            }
          },
        }}
        ref={inputRef}
        inputProps={{ ref: null }}
        component={DateInput}
        classNames={styles}
        format={format}
        placeholder={placeholder}
        formatDate={getDate}
        parseDate={parseDate}
        onDayChange={onDayChange}
        value={value}
      />
    </div>
  );
};

Datepicker.defaultProps = {
  format: DEFAULT_FORMAT,
  placeholder: 'DD.MM.YYYY',
  value: '',
  label: '',
  className: '',
  style: {},
};

export default Datepicker;
