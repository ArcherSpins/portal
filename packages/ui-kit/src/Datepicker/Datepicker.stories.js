/* eslint-disable */
import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react'
import Datepicker from '.';


storiesOf('Datepicker', module)
  .addWithChapters('default', {
    chapters: [
      {
        sections: [
          {
            sectionFn: () => (
              <div style={{ width: '300px'}}>
                <Datepicker onDayChange={action('day changed!')} />
              </div>
            ),
          },
          {
            sectionFn: () => (
              <div style={{ width: '300px'}}>
                <Datepicker value={new Date()} onDayChange={action('day changed!')} />
              </div>
            )
          },
          {
            title: 'With overlayAlign=left',
            sectionFn: () => (
              <div style={{ width: '300px'}}>
                <Datepicker overlayAlign="left" value={new Date()} onDayChange={action('day changed!')} />
              </div>
            )
          }
        ],
      },
    ],
  });