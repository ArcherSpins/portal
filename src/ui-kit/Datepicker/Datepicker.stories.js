/* eslint-disable */
import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react'
import Datepicker from 'ui-kit/Datepicker';


storiesOf('Datepicker', module)
  .addWithChapters('default', {
    chapters: [
      {
        sections: [
          {
            sectionFn: () => (
              <div style={{ width: '300px'}}>
                <Datepicker onDayChange={action('day changed!')} />
              </div>
            ),
          },
          {
            sectionFn: () => (
              <div style={{ width: '300px'}}>
                <Datepicker value={new Date()} onDayChange={action('day changed!')} />
              </div>
            )
          }
        ],
      },
    ],
  });