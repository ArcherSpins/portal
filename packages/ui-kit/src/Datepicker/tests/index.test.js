// TODO: TDD
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';

import Datepicker from '..';

const component = <Datepicker />;

describe('The Datepicker component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(component);
    expect(wrapper.hasClass('datepicker')).toBeTruthy();
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
