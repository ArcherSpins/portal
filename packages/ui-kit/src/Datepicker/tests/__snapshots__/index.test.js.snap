// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Datepicker component renders correctly 1`] = `
<div
  className="datepicker"
  style={Object {}}
>
  <label>
    
  </label>
  <div
    className="container"
  >
    <div
      className="input"
    >
      <div
        className="default md"
      >
        <label>
          
        </label>
        <div
          className="wrap"
        >
          
          <input
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            placeholder="DD.MM.YYYY"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="icon__wrap"
        onBlur={[Function]}
        onClick={[Function]}
        type="button"
      >
        <i
          className="icon-calendar"
        />
      </button>
    </div>
  </div>
</div>
`;
